org.lflang/src/org/lflang/generator/cpp/CppTypes.kt

```diff
@@ -52,67 +49,36 @@ object CppTypes : TargetTypes {
             else magnitude.toString() + unit.cppUnit
         }
 
-}
-
-/**
- * Returns the variable initializer used in ctors. Eg returns `fieldName(1)`.
- */
-fun CppTypes.getCppInitializerList(state: StateVar): String {
-    val init: Initializer? = state.init
-    val inferredType: InferredType = state.inferredType
-    return buildString {
-        append(state.name)
-        if (init == null) {
-            append("/*missing initializer*/")
-        } else {
-            val (prefix, postfix) = if (init.isBraces) Pair("{", "}") else Pair("(", ")")
-            init.exprs.joinTo(this, ", ", prefix, postfix) {
-                getTargetExpr(it, inferredType.componentType)
-            }
-        }
+    override fun getTargetParamRef(expr: ParameterReference, type: InferredType?): String {
+        return "parameters.${expr.parameter.name}"
     }
 }
 
-/** Returns an initializer in the form `int(2)` or `Class {1,2}`. */
-fun CppTypes.getCppStandaloneInitializer(init: Initializer?, inferredType: InferredType): String {
-    if (init == null) {
-        return getMissingExpr(inferredType)
+/** Get a C++ representation of a LF unit. */
+val TimeUnit?.cppUnit
+    get() = when (this) {
+        TimeUnit.NANO   -> "ns"
+        TimeUnit.MICRO  -> "us"
+        TimeUnit.MILLI  -> "ms"
+        TimeUnit.SECOND -> "s"
+        TimeUnit.MINUTE -> "min"
+        TimeUnit.HOUR   -> "h"
+        TimeUnit.DAY    -> "d"
+        TimeUnit.WEEK   -> "d*7"
+        else            -> ""
     }
 
-    return buildString {
-        append(getTargetType(inferredType)) // treat as ctor call
+/**
+ * Returns a C++ variable initializer.
+ */
+fun CppTypes.getCppInitializer(init: Initializer?, inferredType: InferredType): String {
```

Comment:
Will this work out of the box with = initializers? Will we just have to make this return `= init.exprs[0]` basically?

Reply:
Yes, for most cases this should suffice.